checks.d/docker_daemon.py

```diff
@@ -432,13 +436,24 @@ def refresh_ecs_tags(self):
         ip = ecs_config.get('NetworkSettings', {}).get('IPAddress')
         ports = ecs_config.get('NetworkSettings', {}).get('Ports')
         port = ports.keys()[0].split('/')[0] if ports else None
+        if not ip and DockerUtil.is_dockerized():
+            if self.docker_gateway:
+                ip = self.docker_gateway
+                port = ECS_INTROSPECT_DEFAULT_PORT
+        elif not ip:
+            ip = "localhost"
+            port = ECS_INTROSPECT_DEFAULT_PORT
```

Comment:
You could refactor the conditions to look like:
```python
if not ip:
    port = ECS_INTROSPECT_DEFAULT_PORT
    if DockerUtil.is_dockerized() and self.docker_gateway:
        ip = self.docker_gateway
    else:
        ip = 'localhost'

```
which is 1 line shorter \o/ (kidding aside I think it's also a bit clearer)